src/Device/Driver/AR62xx.cpp

```diff
@@ -0,0 +1,670 @@
+  //! Mutex to be locked to access response.
+  std::unique_lock<Mutex> response_mutex;
+  // RadioPara aus LK8000
+  Radio_t RadioPara;
+
+public:
+  /**
+   * Constructor of the radio device class.
+   *
+   * @param _port Port the radio is connected to.
+   */
+  AR62xxDevice(Port &_port);
+
+private:
+  /**
+   * Sends a message to the radio.
+   *
+   * @param msg Message to be send to the radio.
+   */
+  bool Send(const uint8_t *msg, unsigned msg_size, OperationEnvironment &env);
+
+  /*
+   * Called by "PutActiveFrequency"
+   * doing the real work
+   */
+  bool AR620xPutFreqActive(double Freq, const TCHAR *StationName, OperationEnvironment &env);
+
+  /*
+   * Called by "PutStandbyFrequency"
+   * doing the real work
+   */
+  bool AR620xPutFreqStandby(double Freq, const TCHAR *StationName, OperationEnvironment &env);
+
+  /*
+   * Creates the correct index for a given readable frequency
+   */
+  uint16_t Frq2Idx(double fFreq);
+
+  /*
+   * Creates a correct frequency-number given by the index
+   */
+  double Idx2Freq(uint16_t uFreqIdx);
+
+  /*
+   * This function sets the station name and frequency on the AR62xx
+   */
+  int SetAR620xStation(uint8_t *Command, int Active_Passive, double fFrequency, const TCHAR* Station);
+
+  /*
+   * Parses the messages which XCSoar receives from the radio.
+   */
+  bool AR620xParseString(const char *String, size_t len);
+
+  int AR620x_Convert_Answer(uint8_t  *szCommand, int len, uint16_t CRC);
+
+  uint8_t toHexString(const void* data, size_t size);
+
+public:
+  /**
+   * Sets the active frequency on the radio.
+   */
+  virtual bool PutActiveFrequency(RadioFrequency frequency,
+                                  const TCHAR *name,
+                                  OperationEnvironment &env) override;
+  /**
+   * Sets the standby frequency on the radio.
+   */
+  virtual bool PutStandbyFrequency(RadioFrequency frequency,
+                                   const TCHAR *name,
+                                   OperationEnvironment &env) override;
+  /**
+   * Receives and handles data from the radio.
+   *
+   * The function parses messages send by the radio.
+   * Because all control characters (e.g. HEADER_ID, PROTOKOLL_ID, STX, ACK, NAK, ...)
+   * can be part of the payload of the messages, it is important
+   * to separate the messages to distinguish control characters
+   * from payload characters.
+   *
+   * If a response to a command is received, the function notifies
+   * the sender. This could trigger a retransmission in case of a
+   * failure.
+   */
+  virtual bool DataReceived(const void *data, size_t length, struct NMEAInfo &info) override;
+};
+
+// Little helpers for me
+//size_t = long unsigned int
+//unit16_t = unsigned short int
+//uint8_t = unsigned char
+//TCHAR = char
+
+/*
+ * Converting to a hex string for a useful output via LogFormat
+ */
+uint8_t AR62xxDevice::toHexString(const void* data, size_t size) {
+  uint8_t szHex = 0;
+  constexpr TCHAR hex_chars[16] = {
```

Comment:
Missing `static`?
We already have several copies of this kind of code, and I'll not merge yet another. This belongs in a central library.

Reply:
removed